compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/DecodeV1.hs

```diff
@@ -165,97 +165,103 @@ decodeChoice LF1.TemplateChoice{..} =
+  LF1.BuiltinFunctionEQUAL_DATE -> pure $ BEEqual BTDate
+  LF1.BuiltinFunctionEQUAL_PARTY -> pure $ BEEqual BTParty
+  LF1.BuiltinFunctionEQUAL_BOOL -> pure $ BEEqual BTBool
+
+  LF1.BuiltinFunctionLEQ_INT64 -> pure $ BELessEq BTInt64
+  LF1.BuiltinFunctionLEQ_NUMERIC -> pure $ BELessEq BTDecimal
+  LF1.BuiltinFunctionLEQ_TEXT    -> pure $ BELessEq BTText
+  LF1.BuiltinFunctionLEQ_TIMESTAMP    -> pure $ BELessEq BTTimestamp
+  LF1.BuiltinFunctionLEQ_DATE -> pure $ BELessEq BTDate
+  LF1.BuiltinFunctionLEQ_PARTY -> pure $ BELessEq BTParty
+
+  LF1.BuiltinFunctionLESS_INT64 -> pure $ BELess BTInt64
+  LF1.BuiltinFunctionLESS_NUMERIC -> pure $ BELess BTDecimal
+  LF1.BuiltinFunctionLESS_TEXT    -> pure $ BELess BTText
+  LF1.BuiltinFunctionLESS_TIMESTAMP    -> pure $ BELess BTTimestamp
+  LF1.BuiltinFunctionLESS_DATE -> pure $ BELess BTDate
+  LF1.BuiltinFunctionLESS_PARTY -> pure $ BELess BTParty
+
+  LF1.BuiltinFunctionGEQ_INT64 -> pure $ BEGreaterEq BTInt64
+  LF1.BuiltinFunctionGEQ_NUMERIC -> pure $ BEGreaterEq BTDecimal
+  LF1.BuiltinFunctionGEQ_TEXT    -> pure $ BEGreaterEq BTText
+  LF1.BuiltinFunctionGEQ_TIMESTAMP    -> pure $ BEGreaterEq BTTimestamp
+  LF1.BuiltinFunctionGEQ_DATE -> pure $ BEGreaterEq BTDate
+  LF1.BuiltinFunctionGEQ_PARTY -> pure $ BEGreaterEq BTParty
+
+  LF1.BuiltinFunctionGREATER_INT64 -> pure $ BEGreater BTInt64
+  LF1.BuiltinFunctionGREATER_NUMERIC -> pure $ BEGreater BTDecimal
+  LF1.BuiltinFunctionGREATER_TEXT    -> pure $ BEGreater BTText
+  LF1.BuiltinFunctionGREATER_TIMESTAMP    -> pure $ BEGreater BTTimestamp
+  LF1.BuiltinFunctionGREATER_DATE -> pure $ BEGreater BTDate
+  LF1.BuiltinFunctionGREATER_PARTY -> pure $ BEGreater BTParty
+
+  LF1.BuiltinFunctionTO_TEXT_INT64 -> pure $ BEToText BTInt64
+  LF1.BuiltinFunctionTO_TEXT_NUMERIC -> pure $ BEToText BTDecimal
+  LF1.BuiltinFunctionTO_TEXT_TEXT    -> pure $ BEToText BTText
+  LF1.BuiltinFunctionTO_TEXT_TIMESTAMP    -> pure $ BEToText BTTimestamp
+  LF1.BuiltinFunctionTO_TEXT_PARTY   -> pure $ BEToText BTParty
+  LF1.BuiltinFunctionTO_TEXT_DATE -> pure $ BEToText BTDate
+  LF1.BuiltinFunctionTEXT_FROM_CODE_POINTS -> pure BETextFromCodePoints
+  LF1.BuiltinFunctionFROM_TEXT_PARTY -> pure BEPartyFromText
+  LF1.BuiltinFunctionFROM_TEXT_INT64 -> pure BEInt64FromText
+  LF1.BuiltinFunctionFROM_TEXT_NUMERIC -> pure BEDecimalFromText
+  LF1.BuiltinFunctionTEXT_TO_CODE_POINTS -> pure BETextToCodePoints
+  LF1.BuiltinFunctionTO_QUOTED_TEXT_PARTY -> pure BEPartyToQuotedText
+
+  LF1.BuiltinFunctionADD_NUMERIC   -> pure BEAddDecimal
+  LF1.BuiltinFunctionSUB_NUMERIC   -> pure BESubDecimal
+  LF1.BuiltinFunctionMUL_NUMERIC   -> pure BEMulDecimal
+  LF1.BuiltinFunctionDIV_NUMERIC   -> pure BEDivDecimal
+  LF1.BuiltinFunctionROUND_NUMERIC -> pure BERoundDecimal
+  LF1.BuiltinFunctionCAST_NUMERIC  ->
+    -- FixMe https://github.com/digital-asset/daml/issues/2289
+    throwError $ ParseError "builtin CAST_NUMERIC not supported"
```

Comment:
Let's use `error` here instead and leave the `pure` at the top. All of this is sitting behind a checker which should make sure we never hit this case anyway.

Reply:
I just remove the "new" builtins from the proto (We will add builtins, once we exactly know the ones we need).  Then I can put back the `pure` at the top, without using `error` .